### PR TITLE
Callback function typings for PushNotificationIOS

### PR DESCRIPTION
**Pull request by RalfNieuwenhuizen**
Allow the different kind of callback functions for different events in the `PushNotificationIOS` class, according to the specification in https://facebook.github.io/react-native/docs/pushnotificationios.html#addeventlistener

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://facebook.github.io/react-native/docs/pushnotificationios.html#addeventlistener

### CODE REVIEW

**Review by timwangdev — APPROVED** [review]
LGTM

File: types/react-native/index.d.ts [hunk 1]
```diff
@@ -7288,13 +7288,36 @@ export interface PushNotificationIOSStatic {
      *
      * The type MUST be 'notification'
```
Comment (Pajn) [inline types/react-native/index.d.ts]: This comment (starting with "remote notifications") should probably be updated to include local notifications or just dropped.

**Review by Pajn — CHANGES_REQUESTED** [review]

File: types/react-native/index.d.ts (hunk 1 above)
Comment (RalfNieuwenhuizen) [reply types/react-native/index.d.ts]: @Pajn if you check https://facebook.github.io/react-native/docs/pushnotificationios.html#addeventlistener you see I copied (and slightly modified to fit format) their comments. Official documentation says "remote".

Comment (timwangdev) [reply types/react-native/index.d.ts]: We can just drop it. a lot of docs in react-native website are outdated.

Comment (Pajn) [reply types/react-native/index.d.ts]: I see. I still think the `The type MUST be 'notification'` should either match the type or be dropped though.

I shall not block the merge though. I will see if I can make @typescript-bot happy again

**Review by Pajn — COMMENTED** [review]
Will this remove the block?

**Review by Pajn — APPROVED** [review]
Nope. Maybe this?

(sorry for the noise!)

**Pull request merged**